File: hash/349.两个数组的交集.java
/*
 * @lc app=leetcode.cn id=349 lang=java
 *
 * [349] 两个数组的交集
 */

// @lc code=start
class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {
        Set<Integer> tempMap = new HashSet<>();
        
        int k = 0;

        for (int num : nums1) {
            if (!tempMap.contains(num)) {
                tempMap.add(num);
            }
        }
        List<Integer> resList = new ArrayList<>();
        for (int num : nums2) {
            if (tempMap.contains(num)) {
                resList.add(num);
                tempMap.remove(num);
            }
        }

        int[] res = new int[resList.size()];

        for (int i = 0; i < resList.size(); i++) {
            res[i] = resList.get(i);
        }

        return res;
    }
}
// @lc code=end

File: hash/349.两个数组的交集.py
#
# @lc app=leetcode.cn id=349 lang=python3
#
# [349] 两个数组的交集
#

# @lc code=start
class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        tempSet = set()
        res = []

        for num in nums1:
            tempSet.add(num)

        for num in nums2:
            if num in tempSet:
                res.append(num)
                tempSet.discard(num)

        return res
# @lc code=end

